plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -483,6 +483,7 @@ func (h *ConsumerGroupHandler) Handle(session sarama.ConsumerGroupSession, msg *
 
 	metrics, err := h.parser.Parse(msg.Value)
 	if err != nil {
+		session.MarkMessage(msg, "")
```

Comment:
Should we provide some metadata? Like
```suggestion
		session.MarkMessage(msg, "telegraf: failed parsing")
```

Reply:
I don't think that metadata is meant for random strings or reasons. That metadata is passed to the markoffset call. There is a comment in the docs about other consumers using that string to restore state.

As we do not currently do that for other errors, I would be hesitant to start.